src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -30,112 +36,164 @@ public void addSection(Section section) {
         sectionList.add(section);
     }
 
+    private boolean isAlreadyExistSection(Section section) {
+        return anyMatch(section::equalsUpAndDownStation);
+    }
+
     private boolean nonExistStation(Section section) {
-        return !sectionList.isEmpty() && sectionList.stream()
-                .noneMatch(otherSection ->
+        return !sectionList.isEmpty() && noneMatch(otherSection ->
                         section.getUpStation().equals(otherSection.getUpStation())
                         || section.getUpStation().equals(otherSection.getDownStation())
                         || section.getDownStation().equals(otherSection.getUpStation())
                         || section.getDownStation().equals(otherSection.getDownStation())
                 );
     }
 
-    private boolean isAlreadyExistSection(Section section) {
-        return sectionList.stream()
-                .anyMatch(section::equals);
-    }
-
     private boolean isMiddleSection(Section section) {
-        return sectionList.stream()
-                .anyMatch(otherSection ->
+        return anyMatch(otherSection ->
                         section.getUpStation().equals(otherSection.getUpStation())
                                 || section.getDownStation().equals(otherSection.getDownStation()));
     }
 
     private void addMiddleSection(Section section) {
-        for (Section otherSection : sectionList) {
-            if (section.getUpStation().equals(otherSection.getUpStation())) {
-                validateDistance(section, otherSection);
-                otherSection.changeUpStation(section.getDownStation(), section.getDistance());
-                break;
-            }
-
-            if (section.getDownStation().equals(otherSection.getDownStation())) {
-                validateDistance(section, otherSection);
-                otherSection.changeDownStation(section.getUpStation(), section.getDistance());
-                break;
-            }
+        Optional<Section> sameUpStationSection = findSectionBy(section::hasSameUpStation);
+        if (sameUpStationSection.isPresent()) {
+            Section beforeSection = sameUpStationSection.get();
+            Section changedSection = beforeSection.changeUpStation(section);
+            replaceSection(beforeSection, changedSection);
+            sectionList.add(section);
+            return;
```

Comment:
```suggestion
        findSectionBy(section::hasSameUpStation)
                .ifPresent(foundSection-> {
                    Section changedSection = foundSection.changeUpStation(section);
                    replaceSection(foundSection, changedSection);
                    sectionList.add(section);
                });
```
ifPresent() 메서드를 활용하면 더 간결하게 처리할 수 있을 것 같아요.